prober/icmp.go

```diff
@@ -287,18 +312,23 @@ type v4Conn struct {
 	src net.IP
 }
 
-func (c *v4Conn) ReadFrom(b []byte) (int, net.Addr, error) {
-	h, p, _, err := c.c.ReadFrom(b)
+func (c *v4Conn) ReadFromCM(b []byte) (int, *ipv4.ControlMessage, net.Addr, error) {
+	h, p, cm, err := c.c.ReadFrom(b)
 	if err != nil {
-		return 0, nil, err
+		return 0, nil, nil, err
 	}
 
 	copy(b, p)
 	n := len(b)
 	if len(p) < len(b) {
 		n = len(p)
 	}
-	return n, &net.IPAddr{IP: h.Src}, nil
+	return n, cm, &net.IPAddr{IP: h.Src}, nil
+}
+
+func (c *v4Conn) ReadFrom(b []byte) (int, net.Addr, error) {
```

Comment:
We don't need a trivial wrapper like this, just change the existing function.

Reply:
I could not change the signature as it would violate the interface.